src/processors/default/ipfsMimeTypeProcessor.ts

```diff
@@ -0,0 +1,71 @@
+
+import { Table } from '../../db/db';
+import { IPFSFile } from '../../types/ipfsFile';
+import { MimeEnum } from '../../types/media';
+import { Clients, Processor } from '../../types/processor';
+import { Trigger } from '../../types/trigger';
+import { rollPromises } from '../../utils/rollingPromises';
+
+const TIMEOUT = parseInt(process.env.METADATA_REQUEST_TIMEOUT!)
+const QUERY_LIMIT = process.env.QUERY_TRIGGER_BATCH_SIZE
+
+const missingMimeType: Trigger<undefined> = async (clients) => {
```

Comment:
looks like the current ipfs uploader is actually being limited by the IPFS_UPLOAD_BATCH_SIZE env variable which i have set to just 1

so i'd suggest maybe add another env variable for your HEAD getter that's lower than the global limits and go down to like 10 max items on the trigger

also, trigger only on files after they've been pinned